platform/security/certificate/security-certificate-generator/src/test/java/org/codice/ddf/security/certificate/generator/CertificateCommandTest.java

```diff
@@ -0,0 +1,242 @@
+        final Certificate c = pke.getCertificate();
+        final X509CertificateHolder holder = new X509CertificateHolder(c.getEncoded());
+
+        MatcherAssert.assertThat("There should be no subject alternative name extension",
+                holder.getExtension(org.bouncycastle.asn1.x509.Extension.subjectAlternativeName),
+                nullValue(org.bouncycastle.asn1.x509.Extension.class));
+    }
+
+    private static void validateCertificateHasSANs(KeyStoreFile ksf, String alias)
+            throws Exception {
+        final KeyStore.Entry ke = ksf.getEntry(alias);
+
+        assertThat(ke, instanceOf(KeyStore.PrivateKeyEntry.class));
+        final KeyStore.PrivateKeyEntry pke = (KeyStore.PrivateKeyEntry) ke;
+        final Certificate c = pke.getCertificate();
+        final X509CertificateHolder holder = new X509CertificateHolder(c.getEncoded());
+        final org.bouncycastle.asn1.x509.Extension csn =
+                holder.getExtension(org.bouncycastle.asn1.x509.Extension.subjectAlternativeName);
+
+        MatcherAssert.assertThat(csn.getParsedValue()
+                        .toASN1Primitive()
+                        .getEncoded(ASN1Encoding.DER),
+                equalTo(new GeneralNamesBuilder().addName(new GeneralName(GeneralName.iPAddress,
+                        "1.2.3.4"))
+                        .addName(new GeneralName(GeneralName.dNSName, "A"))
+                        .build()
+                        .getEncoded(ASN1Encoding.DER)));
+    }
+
+    private static void validateKeyStore(String alias, boolean hasSAN) throws Exception {
+        final KeyStoreFile ksf = CertificateCommand.getKeyStoreFile();
+
+        assertThat(ksf.aliases()
+                .size(), is(2));
+        if (hasSAN) {
+            validateCertificateHasSANs(ksf, alias);
+        } else {
+            validateCertificateHasNoSAN(ksf, alias);
+        }
+    }
+
+    @Before
+    public void setup() throws IOException {
+        this.systemKeystoreFile = temporaryFolder.newFile("serverKeystore.jks");
+        final FileOutputStream systemKeyOutStream = new FileOutputStream(systemKeystoreFile);
+        final InputStream systemKeyStream = CertificateGenerator.class.getResourceAsStream(
+                "/serverKeystore.jks");
+
+        IOUtils.copy(systemKeyStream, systemKeyOutStream);
+        IOUtils.closeQuietly(systemKeyOutStream);
+        IOUtils.closeQuietly(systemKeyStream);
+
+        System.setProperty(SecurityConstants.KEYSTORE_TYPE, "jks");
+        System.setProperty(SecurityConstants.KEYSTORE_PATH, systemKeystoreFile.getAbsolutePath());
+        System.setProperty(SecurityConstants.KEYSTORE_PASSWORD, "changeit");
+    }
+
+    @Test
+    public void testConfigureDemoCertWithoutSAN() throws Exception {
+        final KeyStoreFile ksf = CertificateCommand.getKeyStoreFile();
+
+        assertThat(ksf.aliases()
+                .size(), is(2));
+        assertThat(ksf.isKey("my-fqdn"), is(false));
```

Comment:
Hum! These are used here to validate (or assert) the result of the tests below by verifying the content of the generated keystore and not necessarily to test the getKeyStoreFile().